spec/controllers/saml_idp_controller_spec.rb

```diff
@@ -119,6 +119,31 @@
 
       expect(response).to be_bad_request
     end
+
+    context 'cert element in SAML request is blank' do
+      let(:user) { create(:user, :fully_registered) }
+      let(:service_provider) { build(:service_provider, issuer: 'http://localhost:3000') }
+
+      # the RubySAML library won't let us pass an empty string in as the certificate
+      # element, so this test substitutes a SAMLRequest that has that element blank
+      # rubocop:disable Layout/LineLength
+      let(:blank_cert_element_req) do
+        'lVVbc6JMEH3fqv0PFD5ahLsiFd3SECNeYkSNl5etcRhwIgzKDN5+/Qfksuhm98tWUTz0dJ8+faZ7+vbHMQy4PYopjkidl28k/kfj+7dbCsJga/YjP0qYg3YJooxLPQk1s5M6n8TEjADF1CQgRNRk0Bw3B31TuZFMQCmKWYrHF0K2f4/ZxhGLYBTwnJWmwgSwnM+asa0piofD4QYdQbgN0A2MQhFssZiBikFOUJEUledsq87/VBTVlY0KFFQdSILmyp6wMmQgQK2qAlerVTRZT10pTZBNKAOE1fk0XBMkWZDliWyYimJK6pLnnt81SenxqSQcl4ti5rFx441ZEEEQrCPKTFWSpFux6JLHuNQcYz+tJ4nRmxwuvSjsoN5EsS8qabwo1cTUx6XYL73m/IWAXJt40Zvx1XwHSERwSgCfc70GiK0jl2sGfhRjtg7/kEYWZSlLI6AjFKCskRIvXuB+EP4yoKS98xbCKEalmAKBroGiV66gHeShGBGIuKlj1/nS1y7sA+IVZBIDQr0oDmnh4Orof0lfiI3IHgXRFrkCfa+9QPzfwT+R+AItxUPQtAkMEor36DGbhy2AiL61CIJfB+aeYuThYx/TtJdLLvJAEjAun7n8z7kpavZhLnSvqxKLZRVlFv+kcyaEhf10Sv+xO9JbL/3WE0W8ZxAkqKE8rlSnv8LtDpHgiZSr3WZ1PVutevLjvj14fgDaYDFZqaOabNdzlsXgj1YTi732PkniZ6N00fCvKNOQOhDMNp15PCWtgTccte4Uyx5t5Hgz2g2gHiXwXGk3X5JgJy3n1d2kfJ6uxstqT0xoOTjDvryO/c4cK4fjS2/sYExncfdxHK7sh4elDbTy1DjYg/UmWej75XPPPzpQAU+GgcVgdpg6A1s2VOw1z3eH7U5faCfQrWw75ZeVvxiycLiG87U8PMR2ebRzqqOxH5ymXV2fDcutO0eqdjyjWbXGTXHmb2vHTtmaHuR7clhZ7jnC1Ul7BzoLY6yNZzaESsXq0J1hhC80pMdm2xmVNVYhwB2SOG6T7qzXRbgdWMbgCEdJCyeLFmI9p28/HU/zXvv4ck+q95uufrY2m7g1O00n2vNE7BK2nqZBim6V94a8nNtzdeLX6x8XUND61yX00Om3F26uSzULMHDVL5n5LtsxXvr6MdTIga+NF71wBZTbigkvqRWe+2w0bYtrp1MA2N93WGbBruDlribLhgcjwvjGT1kFqCa5sqB4tZWgIVdPHznNFdSqoiCgKUg1Km+74zVftoLFT3Zw4z8='
```

Comment:
is there a way we could have have the spec have the XML in it? and then we do the `base64(deflate(str))` on it?

Reply:
updated in https://github.com/18F/identity-idp/pull/9908/commits/e3b6d06c4473bc8467e7c7c6ebbfbb26a7114604